desc/coils.py

```diff
@@ -77,7 +76,7 @@ def compute_magnetic_field(self, coords, params=None, basis="rpz", grid=None):
             magnetic field at specified points, in either rpz or xyz coordinates
         """
         assert basis.lower() in ["rpz", "xyz"]
-        if isinstance(coords, Grid):
+        if hasattr(coords, "nodes"):
```

Comment:
is `hasattr` instead of `isinstance` better in some way, or is this just a cleaner way? Why this change is what I am asking

Reply:
In this case it is "better" since it's what we actually need from the object. In general I was just trying to avoid importing things just for instance checking where possible